frontend/packages/ceph-storage-plugin/src/components/modals/storage-pool-modal/storage-pool-modal.tsx

```diff
@@ -0,0 +1,391 @@
+import { k8sCreate, k8sGet } from '@console/internal/module/k8s/resource';
+
+import { CephClusterKind, StoragePoolKind, OCSStorageClusterKind } from '../../../types';
+import { CephBlockPoolModel, OCSServiceModel } from '../../../models';
+import { CEPH_STORAGE_NAMESPACE, OCS_INTERNAL_CR_NAME } from '../../../constants/index';
+
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { replicas, modalTitle, modalDescription, notReadyDesc } from './storage-pool-constant';
+
+import './_storage-pool-modal.scss';
+
+const LoadingComponent: React.FC = () => (
+  <span className="pf-c-spinner" role="progressbar" aria-valuetext="Loading...">
+    <span className="pf-c-spinner__clipper" />
+    <span className="pf-c-spinner__lead-ball" />
+    <span className="pf-c-spinner__tail-ball" />
+  </span>
+);
+
+const PoolStatusComponent: React.FC<PoolStatusComponentProps> = ({ status, name, error = '' }) => {
+  return (
+    <div>
+      {status === 'progress' && (
+        <EmptyState>
+          <EmptyStateIcon icon={LoadingComponent} />
+          <EmptyStateBody>Pool {name} creation in progress</EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'created' && (
+        <EmptyState>
+          <EmptyStateIcon icon={CheckCircleIcon} className="ceph-storage-pool__check-icon" />
+          <EmptyStateBody>Pool {name} was successfully created</EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'failed' && (
+        <EmptyState>
+          <EmptyStateIcon icon={ExclamationCircleIcon} className="ceph-storage-pool__error-icon" />
+          <EmptyStateBody>
+            {error || `An error occurred, Pool ${name} was not created`}
+          </EmptyStateBody>
+        </EmptyState>
+      )}
+      {status === 'timeout' && (
+        <EmptyState>
+          <EmptyStateIcon icon={UnknownIcon} className="ceph-storage-pool__timeout-icon" />
+          <EmptyStateBody>
+            Pool creation timed out. Please check the events to know pool {name} status
+          </EmptyStateBody>
+        </EmptyState>
+      )}
+    </div>
+  );
+};
+
+export const StoragePoolModal = withHandlePromise((props: StoragePoolModalProps) => {
+  const {
+    cephClusterObj,
+    onPoolCreation,
+    close,
+    cancel,
+    handlePromise,
+    errorMessage,
+    inProgress,
+  } = props;
+  const [newPoolName, setNewPoolName] = React.useState('sc-pool');
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isPerfObjOpen, setPerfObjOpen] = React.useState(false);
+  const [replicaSize, setReplicaSize] = React.useState('');
+  const [isCompressed, setCompression] = React.useState(false);
+  const [deviceClass, setdeviceClass] = React.useState('');
+  const [poolStatus, setPoolStatus] = React.useState('');
+  const [storageClusterLoading, setStorageClusterLoading] = React.useState(true);
```

Comment:
May be you can create an object like 
```
const storageClustetState = {
   loading: true,
   error: false,
   ready: false,
}
```
and use this whole object as state. It will look more cleaner.

Reply:
ack